Grasshopper.fs

```diff
@@ -293,17 +297,26 @@ module Pretty =
     /// <returns>
     ///     A <see cref="Doc"/> representing <paramref name="cmd"/>.
     /// </returns>
-    let printCommand (cmd : GrassCommand) : Doc =
-        let c =
-            match cmd with
-            | VarDecl var -> String "var" <+> printTypedGrass printShadowVar var
-            | NoAssign prim -> printPrim prim
-            | Assign (lvalue, prim) ->
-                printShadowVar lvalue <+> String ":=" <+> printPrim prim
-            | PureAssume assumption ->
-                String "pure assume"
-                    <+> printBoolExprG (printSymGrass printGrassVar) assumption
-        withSemi c
+    let rec printCommand (cmd : GrassCommand) : Doc =
+        let printBlock c = 
+            braced (ivsep (List.map (printCommand >> Indent) c))
+
+        match cmd with
+        | VarDecl var ->
+            withSemi (String "var" <+> printTypedGrass printShadowVar var)
+        | NoAssign prim ->
+            withSemi (printPrim prim)
+        | Assign (lvalue, prim) ->
+            withSemi (printShadowVar lvalue <+> String ":=" <+> printPrim prim)
+        | PureAssume assumption ->
+            withSemi
+                (String "pure assume"
+                    <+> printBoolExprG (printSymGrass printGrassVar) assumption)
+        | Branch (cond = c; trueBranch = t; falseBranch = f) ->
+            String "if"
+            <+> parened (printBoolExprG (printSymGrass printGrassVar) c)
+            <+> printBlock t
+            <+> (if f.IsEmpty then Nop else (String "else" <+> printBlock f))
```

Comment:
As well as adding `Branch`, rearrange this because not every command now needs `withSemi`.